gtsfm/bundle/bundle_adjustment.py

```diff
@@ -195,6 +260,18 @@ def __optimize_factor_graph(self, graph: NonlinearFactorGraph, initial_values: V
         result_values = lm.optimize()
         return result_values
 
+    def __cameras_to_model(
+        self,
+        initial_data: GtsfmData,
+        absolute_pose_priors: List[Optional[PosePrior]],
+        relative_pose_priors: Dict[Tuple[int, int], Optional[PosePrior]],
```

Comment:
why do we need the priors here?

Reply:
Ideally I'd like some sophisticated logic here, which decides if the prior information is good enough to optimize for a camera, even if its missing in the initial data (due to upstream failures for that camera)